android/app/src/main/kotlin/coop/polypoly/polypod/ZipTools.kt

```diff
@@ -0,0 +1,55 @@
+package coop.polypoly.polypod.polyNav
+
+import android.content.Context
+import androidx.security.crypto.EncryptedFile
+import androidx.security.crypto.MasterKey
+import java.io.File
+import java.io.InputStream
+import java.util.zip.ZipInputStream
+
+class ZipTools {
+    companion object {
+        open fun unzipAndEncrypt(
+            inputZipStream: InputStream,
+            context: Context,
+            folder: String
+        ) {
+            ZipInputStream(inputZipStream).use { zis ->
+                var entry = zis.nextEntry
+                while (entry != null) {
+                    val fileName = entry.name
+                    val filePath = context.filesDir.absolutePath.plus(
+                        "/$folder/$fileName"
```

Comment:
Would this fail if `folder` does not exist? If what I say above does not exist, shouldn't you check previously?

Reply:
This is merely creating a path. The folder is then created below on line 26 so I don't think it makes sense to check if it exists here?